### PR TITLE
fix(action-list): refactor css

### LINKED ISSUE #3614: Action list updates
Reconsider CSS to match flex spacer system. Something like https://codepen.io/mcoker/pen/jOrMxVa would assign everything via a single `margin-left`. Also need to make sure an icon list/group only affects its direct children, and not nested groups.

### PR DESCRIPTION
**Pull request by christiemolloy**
closes #3614

### CODE REVIEW

File: src/patternfly/components/ActionList/action-list.scss [hunk 1]
```diff
@@ -1,27 +1,33 @@
...
 
   + .pf-c-action-list__group {
-    margin-left: var(--pf-c-action-list__group--group--spacer);
+    margin-left: var(--pf-c-action-list--group--spacer);
   }
 
-  + .pf-c-action-list__item {
-    margin-left: var(--pf-c-action-list__group--item--spacer);
+  &.pf-m-icons {
+    --pf-c-action-list--child--spacer: 0;
```
Comment (mattnolting) [inline src/patternfly/components/ActionList/action-list.scss:31]:
```suggestion
    --pf-c-action-list--child--spacer: var(--pf-c-action-list--m-icon--spacer);
```

File: src/patternfly/components/ActionList/action-list.scss [hunk 2]
```diff
@@ -1,27 +1,33 @@
...
+  --pf-c-action-list--child--spacer-base: var(--pf-global--spacer--md);
+  --pf-c-action-list--group--spacer-base: var(--pf-global--spacer--2xl);
+  --pf-c-action-list--child--spacer: var(--pf-c-action-list--child--spacer-base);
+  --pf-c-action-list--group--spacer: var(--pf-c-action-list--group--spacer-base);
 }
 
 .pf-c-action-list,
 .pf-c-action-list__group {
+  --pf-c-action-list--child--spacer: var(--pf-c-action-list--child--spacer-base);
+  --pf-c-action-list--group--spacer: var(--pf-c-action-list--group--spacer-base);
```
Comment (mattnolting) [inline src/patternfly/components/ActionList/action-list.scss:12]: Looks like you've already set this on line 6

**Review by mattnolting — APPROVED** [review]
LPTM! 🌮

**Review by mcoker — APPROVED** [review]

**Pull request merged**